src/main/java/one/util/streamex/MoreCollectors.java

```diff
@@ -168,6 +168,209 @@ private MoreCollectors() {
         }, Function.identity(), set -> set.size() == size, UNORDERED_ID_CHARACTERISTICS);
     }
 
+    /**
+     * Returns a {@code Collector} that accumulates elements into a {@code Map}
+     * whose keys and values are taken from {@code Map.Entry}.
+     *
+     * @param  <K> the {@link Comparable} type of then map keys
+     * @param  <V> the type of the map values
+     *
+     * @return {@code Collector} which collects elements into a {@code Map}
+     * whose keys and values are taken from {@code Map.Entry}
+     * @throws IllegalStateException if this stream contains duplicate keys
+     *         (according to {@link Object#equals(Object)})
+     *
+     * @see Collectors#toMap(Function, Function)
+     */
+    public static <K, V> Collector<Entry<K, V>, ?, Map<K, V>> entriesToMap() {
```

Comment:
I believe, PECS signature (`Collector<Map.Entry<? extends K, ? extends V>, ?, Map<K, V>>`) could be useful. E.g.:

Map<String, String> origMap = new HashMap<>();
Map<CharSequence, CharSequence> result = origMap.entrySet().stream().collect(entriesToMap()); // doesn't compile now